src/libDirectoryService/DirectoryService.cpp

```diff
@@ -1087,4 +1081,28 @@ uint64_t DirectoryService::CalculateNumberOfBlocksPerYear() const {
 int64_t DirectoryService::GetAllPoWSize() const {
   std::lock_guard<mutex> lock(m_mutexAllPOW);
   return m_allPoWs.size();
+}
+
+void DirectoryService::GetEntireNetworkPeerInfo(
+    std::vector<std::pair<PubKey, Peer>>& peers, std::vector<PubKey>& pubKeys) {
+  peers.clear();
+  pubKeys.clear();
+
+  for (const auto& i : *m_mediator.m_DSCommittee) {
+    if (i.second.m_listenPortHost != 0) {
+      peers.emplace_back(i);
+    }
```

Comment:
Can add `continue` and skip adding my own pubkey?

Reply:
Fixed. Adding `continue` is not relevant though.